src/moab/mesh_manager.cpp

```diff
@@ -269,6 +270,78 @@ MOABMeshManager::get_volume_surfaces(MeshID volume) const
   return surface_ids;
 }
 
+std::vector<Vertex> 
+MOABMeshManager::get_surface_vertices(MeshID surface) const
+{
+  moab::Range faces = _surface_faces(surface);
+  moab::Range verts;
+  this->moab_interface()->get_adjacencies(faces, 0, false, verts, moab::Interface::UNION);
+  std::vector<double> coords(verts.size() * 3);
+  this->moab_interface()->get_coords(verts, coords.data());
+  std::vector<Vertex> vertices(verts.size());
+
+  for (int i = 0; i < verts.size(); i++) {
+    vertices[i] = Vertex(coords[3*i], coords[3*i+1], coords[3*i+2]);
+  }
+
+  return vertices; 
+}
```

Comment:
Something like this will allow you to reuse it below...
```suggestion
   return _get_coords(verts);
}

std::vector<Vertex> MOABMeshManager::_get_coords(moab::Range& verts) {

  std::vector<double> coords(verts.size() * 3);
  this->moab_interface()->get_coords(verts, coords.data());
  std::vector<Vertex> vertices(verts.size());

  for (int i = 0; i < verts.size(); i++) {
    vertices[i] = Vertex(coords[3*i], coords[3*i+1], coords[3*i+2]);
  }

  return vertices; 
}
```

Reply:
I think this is a nice way to do things, so I'll go with this moving forwards.